### PR TITLE
Add Equal() method to Tags

### PR DESCRIPTION
**Pull request by prateek**

### CODE REVIEW

File: ident/tags_test.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	tagsB := Tags{
+		StringTag("foo", "bar"),
+		StringTag("and", "done"),
+	}
+
+	require.False(t, tagsA.Equal(tagsB))
+	require.False(t, tagsB.Equal(tagsA))
+}
+
+func TestTagsInequalOrder(t *testing.T) {
```
Comment (jeromefroe) [inline ident/tags_test.go]: nit: s/TestTagsInequalOrder/TestTagsUnequalOrder

File: ident/types.go [hunk 2]
```diff
@@ -143,6 +143,28 @@ func (ids IDs) Finalize() {
...
 
+// Finalize finalizes all Tags.
+func (tags Tags) Finalize() {
+	for _, t := range tags {
+		t.Name.Finalize()
+		t.Value.Finalize()
+	}
+}
+
+// Equal returns a bool indicating if the tags are equal.
```
Comment (jeromefroe) [inline ident/types.go]: nit: perhaps it's worth calling out that the tags must be in the same order?

File: ident/tags_test.go (hunk 1 above)
Comment (prateek) [reply ident/tags_test.go]: done

File: ident/types.go (hunk 2 above)
Comment (prateek) [reply ident/types.go]: sure thing

**Pull request merged**